subprojects/logging/src/main/java/org/gradle/internal/deprecation/DeprecationLogger.java

```diff
@@ -20,31 +20,32 @@
 import org.gradle.internal.featurelifecycle.DeprecatedUsageBuildOperationProgressBroadcaster;
 import org.gradle.internal.featurelifecycle.LoggingDeprecatedFeatureHandler;
 import org.gradle.internal.featurelifecycle.UsageLocationReporter;
+import org.gradle.util.GradleVersion;
 
 import javax.annotation.CheckReturnValue;
 import javax.annotation.Nullable;
 import javax.annotation.concurrent.ThreadSafe;
 
-import static org.gradle.internal.deprecation.Messages.thisBehaviourHasBeenDeprecatedAndIsScheduledToBeRemoved;
-import static org.gradle.internal.deprecation.Messages.thisIsScheduledToBeRemoved;
-import static org.gradle.internal.deprecation.Messages.thisWillBecomeAnError;
-import static org.gradle.internal.deprecation.Messages.xHasBeenDeprecated;
-
 
 /**
  * Provides entry points for constructing and emitting deprecation messages.
- * The basic deprecation message structure is "Summary. RemovalDetails. Context. Advice. Documentation."
+ * The basic deprecation message structure is "Summary. DeprecationTimeline. Context. Advice. Documentation."
  *
  * The deprecateX methods in this class return a builder that guides creation of the deprecation message.
- * Summary and RemovalDetails parts are populated by the deprecateX methods in this class.
+ * Summary is populated by the deprecateX methods in this class.
  * Context can be added in free text using {@link DeprecationMessageBuilder#withContext(String)}.
  * Advice is constructed contextually using {@link DeprecationMessageBuilder.WithReplacement#replaceWith(Object)} methods based on the thing being deprecated. Alternatively, it can be populated using {@link DeprecationMessageBuilder#withAdvice(String)}.
- * Documentation reference is added using one of:
- *  - {@link DeprecationMessageBuilder#withUpgradeGuideSection(int, String)}
- *  - {@link DeprecationMessageBuilder#withDslReference(Class, String)}
- *  - {@link DeprecationMessageBuilder#withUserManual(String, String)}
+ * DeprecationTimeline is mandatory and is added using one of:
+ * - ${@link DeprecationMessageBuilder#willBeRemovedInVersion(GradleVersion)}
+ * - ${@link DeprecationMessageBuilder#willBeRemovedInNextMajorVersion()}
+ * - ${@link DeprecationMessageBuilder#willBecomeAnError(GradleVersion)}
+ * - ${@link DeprecationMessageBuilder#willBecomeAnErrorInNextMajorVersion()}
```

Comment:
We should simplify/clarify this a bit:

- We don't need the GradleVersion taking methods just yet.  For now, we'll only target major version bumps (7.0, 8.0, etc).
- We don't want the methods to be relative (e.g., next major version).  We want it to be an absolute version.  

So instead of `willBeRemovedInNextMajorVersion`, we can do `willBeRemovedInGradle7`.  We don't need this to be super flexible (e.g., `willBeRemovedInGradle(int)`) because we'll only ever need 1-2 of these methods.  By having a specific method name, it's also easier to find all usages and we don't have to worry about people specifying a version we don't intend.